design/authenticators/authn_oidc/oidc_configuration_enhancements.md

```diff
@@ -0,0 +1,228 @@
+OpenSSL.debug = true
+```
+Running the above in an initializer and will enable debug logging on the OpenIDConnect gem and all its 
+relevant dependencies and ensure that they get routed to the Rails logger. We could also include the debug 
+logs for OpenSSL if we need to specifically focus on certificate-related debugging, which seems to be a 
+common sticking point for authn-oidc users.
+
+### Components
+1. OpenIDConnect - primary gem for [generating an OIDC client](https://github.com/cyberark/conjur/tree/master/app/domain/authentication/authn_oidc/v2/client.rb#L106C20-L106C20) in 
+authn-oidc which can retrieve authorization codes and tokens from the OIDC provider. It outsources discovery 
+to a couple other small gems which we need to consider as well:
+1. SWD (Simple Web Discovery) - handles OIDC configuration discovery against the provider URI configuration 
+endpoint: `/.well-known/openid-configuration` 
+1. WebFinger - used for dynamic discovery when the credential issuer is unknown to the client. **Not used in Conjur.**
+
+### Considerations around OIDC Discovery
+The SWD gem discussed previously is used by OpenIDConnect across many authenticators in Conjur (authn-oidc, 
+authn-jwt, authn-azure, authn-gcp). It performs a couple common tasks in Conjur by invoking the OIDC discovery 
+endpoint:
+  - validate authenticator status and provider connectivity
+  - fetch provider keys for decoding tokens
+
+This should not interfere with the proposed solution, but it is important to note since the modified code will be 
+used across different authenticators to invoke the provider discovery endpoint `/.well-known/openid-configuration`. 
+It will be necessary that the other authenticators are unaffected by this change.
+
+### Testing
+The testing will depend on the final implementation, but at the very least the wrapper method will need to be 
+well-covered by unit tests.
+
+#### Unit tests (temporary truststore modifications)
+1. OpenSSL truststore contains the CA cert symlink while yielding
+1. OpenSSL truststore symlink is removed after yielding
+1. Hash collision in truststore does not produce an error but instead increments the file extension of the symlink
+1. Temp file/dir exists while yielding
+1. Temp file/dir contains expected certificate content while yielding
+1. Temp file/dir is removed after yielding
+
+#### Unit tests (authenticator data object)
+1. Default `ca-cert` value is nil
+1. Set value matches expected
+
+#### Cucumber tests
+One or more of our Keycloak OIDC scenarios should be updated to use a `ca-cert` value provided via Conjur 
+variable. It may turn out to be easier to update all keycloak examples to use the cert variable, and the case 
+where the variable is not set can be covered by the Okta examples.
+
+### Open Questions
+- Do we need validation that `ca-cert` is a valid cert before attempting to use it?
+- Should debug logging of OIDC be enabled by default when Conjur is running in debug?
```

Comment:
Is that what we are doing?

Reply:
I mentioned this to @jtuttle and it sounds like it is preferable to **not** introduce a new configuration, so yes it would be based on whether `CONJUR_LOG_LEVEL` is set to `debug`.

Reply:
Agreed! Our docs provide guidance to toggle Conjur into and out of debug mode while troubleshooting an authenticator.

Reply:
Are these log messages we control? Either way, we need to be certain we don't inadvertently cause disclosures in debug mode.

Reply:
@gl-johnson There is a current issue in Jira that OIDC does not give back stack trace for error logging / troubleshooting
https://ca-il-jira.il.cyber-ark.com:8443/browse/CNJR-2105

Reply:
These would be log messages outside our control. 

OpenIdConnect and SWD logs will contain debug messaging mainly about the HTTP requests to the OIDC provider. The OpenSSL logs may also be useful for lower level issues. When working on a customer case where they had TLS issues in authn-oidc, the OpenSSL logs helped us discover that the CA file they added manually to Conjur did not have the correct permissions.

At most I would expect these logs to include the filepath where OpenSSL expects the CA cert to be, and possibly a public key.